### ISSUE TITLE
Does the training loss include the image-text loss during distillation with CLIP?

### ISSUE THREAD

**Issue by YangC-1** [NVlabs/RADIO#184]
Hi. For distillation with multi teachers including CLIP or SigLIP?  Does AM-RADIO also compute the image2text loss?
something like the following:

s_im_feat = student(im)
text_feat = teacher(text)
logits_per_image = s_im_feat @ text_feat.T
im2text_loss = cross_entropy_loss(logits_per_image, label)

If not, will adding the contrastive loss be helpful for distilling the zero-shot ability?

**Comment by mranzinger** [comment]
Hi. We purely match the teachers vision summary token. That token is itself aligned with the teacher's language model, so we transitively become aligned to the frozen language model.

Adding the contrastive loss is perhaps useful, especially if you are targeting a language domain that isn't well covered by the teacher's training (e.g. medical, etc.). It does presuppose that you have good-ish image+caption pairs though.
